src/VecSim/algorithms/hnsw/hnsw_tiered.h

```diff
@@ -5,15 +5,52 @@
 #include "hnsw_factory.h"
 
 #include <unordered_map>
+/**
+ * Definition of a job that inserts a new vector from flat into HNSW Index.
+ */
+struct HNSWInsertJob : public AsyncJob {
+    void *index;
+    labelType label;
+    idType id;
+
+    HNSWInsertJob(std::shared_ptr<VecSimAllocator> allocator, labelType label_, idType id_,
+                  JobCallback insertCb, void *index_)
+        : AsyncJob(allocator, HNSW_INSERT_VECTOR_JOB, insertCb), index(index_), label(label_),
+          id(id_) {}
+};
+
+/**
+ * Definition of a job that swaps last id with a deleted id in HNSW Index after delete operation.
+ */
+struct HNSWSwapJob : public AsyncJob {
+    void *index;
+    idType deleted_id;
+    long pending_repair_jobs_counter; // number of repair jobs left to complete before this job
+                                      // is ready to be executed (atomic counter).
+    // TODO: implement contractor
+};
+
+/**
+ * Definition of a job that repairs a certain node's connection in HNSW Index after delete
+ * operation.
+ */
+struct HNSWRepairJob : public AsyncJob {
+    void *index;
+    idType node_id;
+    unsigned short level;
+    HNSWSwapJob *associated_swap_job;
```

Comment:
Didn't we say that a repair job should be associated with more than one swap job?

Reply:
No, it's the other way around (multiple repair jobs are associated with a single swap job)